helix-term/src/ui/picker.rs

```diff
@@ -408,21 +126,27 @@ pub struct Picker<T: Item> {
     // pattern: String,
     prompt: Prompt,
     previous_pattern: (String, FuzzyQuery),
-    /// Whether to truncate the start (default true)
-    pub truncate_start: bool,
     /// Whether to show the preview panel (default true)
     show_preview: bool,
     /// Constraints for tabular formatting
     widths: Vec<Constraint>,
 
     callback_fn: Box<dyn Fn(&mut Context, &T, Action)>,
+
+    pub truncate_start: bool,
+    /// Caches paths to documents
+    preview_cache: HashMap<PathBuf, CachedPreview>,
+    read_buffer: Vec<u8>,
+    /// Given an item in the picker, return the file path and line number to display.
+    file_fn: Option<FilePreviewFn<T>>,
 }
 
-impl<T: Item> Picker<T> {
-    pub fn new(
+impl<T: Item + 'static> Picker<T> {
+    fn new(
         options: Vec<T>,
         editor_data: T::Data,
-        callback_fn: impl Fn(&mut Context, &T, Action) + 'static,
+        callback_fn: Box<dyn Fn(&mut Context, &T, Action) + 'static>,
+        preview_fn: Option<FilePreviewFn<T>>,
```

Comment:
As `Picker` is generic now File prefix can be renamed from `FilePreviewFn`?

Reply:
Currently the preview only works for files so I think this name fits. This type also didn't exist before this change

Reply:
#5492 changes that so that a preview of an arbitrary `String`/`Rope` is possible too, maybe just calling it `PreviewFn` now would be easy enough so we don't have to rename in the future?